gen/rules/src/main/scala/clue/gen/Generator.scala

```diff
@@ -188,6 +189,38 @@ trait Generator {
           )
       }
 
+  protected[this] def addOpaqueDef(
+    name:      String,
+    pars:      List[Term.Param],
+    extending: Option[String] = None
+  ): List[Stat] => (List[Stat], Boolean) =
+    parentBody =>
+      mustDefineType(name)(parentBody) match {
+        case Skip                        =>
+          (parentBody, false)
+        case Define(newParentBody, _, _) =>
+          import scala.meta.dialects.Scala3
+
+          val stats: List[Stat] = newParentBody ++ List(
+            Defn.Type(
+              List(Mod.Opaque()),
+              Type.Name(name),
+              Nil,
+              extending.fold[Type](t"_root_.io.circe.Json")(name => Type.Name(name)),
+              Type.Bounds(None, extending.map(name => Type.Name(name)))
+            )
```

Comment:
Can't use quasi-quotes here unfortunately.
- https://github.com/scalameta/scalameta/issues/2881

Reply:
Understood. Also, bummer. Maybe add a comment with this?

Reply:
I hadn't followed-up on the issue, looks like it may be fixed soon (or possibly already in the latest release). Will check.